### PR TITLE
Fix Error/Warning to correcly row

### PR DESCRIPTION
**Pull request by lailton**
I found a small bug to detect the correctly ROW to Warnings message.
This commit solve the problem.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "linter-harbour",
   "main": "./lib/main",
-  "version": "6.6.1",
+  "version": "6.6.2",
```
Comment (Arcanemagus) [inline package.json]: Any version number change will be done when the release is made, please remove this.

File: lib/main.coffee [hunk 2]
```diff
@@ -104,5 +104,28 @@ module.exports =
               #  range[0][0] = Math.min( range[0][0], textEditor.getLineCount()-1)
               #  range[1][0] = Math.min( range[1][0], textEditor.getLineCount()-1)
 
+            regex = /([\w\.]+)\((\d+)\) (Warning) ([\w\d]+) (.+) (\'(.*\((\d+)\))\'|.+)/g
+            #console.log 'output:', output
+            while((match = regex.exec(output)) isnt null)
+              # console.log "match:a", match, "range",
```
Comment (Arcanemagus) [inline lib/main.coffee]: Take out this commented out code, it looks to be leftover from debugging.

**Comment by Arcanemagus** [comment]
Can you provide some example output and how each regex parses it? For example a https://regex101.com/ setup for each of them on the same output?

I'm hoping this can be simplified into a single regex as that would be much preferable.

**Comment by lailton** [comment]
Yes,

Here is a sample with returns for warnings

Harbour 3.2.0dev (r1612161005)
Copyright (c) 1999-2016, http://harbour-project.org/
Compiling 'test.prg'...
test.prg(13) Warning W0003  Variable 'CVAR' declared but not used in function 'MAIN(4)'
test.prg(13) Warning W0003  Variable 'NVAR' declared but not used in function 'MAIN(5)'
test.prg(13) Warning W0032  Variable 'CSTRING' is assigned but not used in function 'MAIN(6)'
test.prg(16) Warning W0001  Ambiguous reference 'SELF'
test.prg(17) Warning W0003  Variable 'A' declared but not used in function 'NAMEFN(13)'
test.prg(17) Warning W0003  Variable 'B' declared but not used in function 'NAMEFN(13)'
test.prg(17) Warning W0003  Variable 'C' declared but not used in function 'NAMEFN(13)'
Lines 752, Functions/Procedures 3
Generating C source output to 'test.c'... Done.

**Comment by Arcanemagus** [comment]
Can you post `test.prg` so I can try this out locally?

**Comment by Arcanemagus** [comment]
From what I can tell so far though, your regex is changing it from using the line number specified at the start, to using the _reference_ line number. This isn't exactly correct though, at most this information should be used to generate a `Trace` referencing where the function that caused the message starts, but the output says the issue is on the line at the start.

Is that not correct? Maybe this is a bug better suited for `harbour` itself.

**Comment by lailton** [comment]
test.prg

#include "hbclass.ch"

function main()
 local cVar
 local nVar
 local cString := "test"

return nil

function nameFn(a,b,c)

return self


It is not a bug, It is how warning is displayed on harbour, It is not the same that error.

Regards,
Lailton Fernando Mariano

**Pull request closed without merging**